edu-services/gradebook-service/api/src/java/org/sakaiproject/service/gradebook/shared/GradebookFrameworkService.java

```diff
@@ -79,4 +80,31 @@
 	 */
 	public void setDefaultGradingScale(String uid);
 
+	/**
+	 *
+	 *	returns all the Available Grading Scales in the system.
+	 */
+	public List getAvailableGradingScales();
+
+	/**
+	 * Adds a new grade scale to an existing gradebook.
+	 *
+	 * @param scaleUuid
+	 *   The Uuid of the scale we want to be added to the gradebook
+	 * @param gradebook
+	 *   The gradebook with GradeMappings where we will add the grading scale.
+	 *
+	 */
+	public void saveGradeMappingToGradebook(String scaleUuid, Object gradebook);
+
+
+	/**
+	 * Update a grademapping with new values.
+	 *
+	 * @param gradeMapping
+	 *   The gradeMApping to update.
+	 *
+	 */
+	public void updateGradeMapping(Object gradeMapping);
```

Comment:
I would be following the other patterns of creating a DTO, eg shared.Assignment, shared.CategoryDefinition etc for passing data around, then mapping it between the DTO and hibernate objects.